examples/example-gateway/app.go

```diff
@@ -39,6 +41,11 @@ var AppOptions = &zanzibar.Options{
 	JSONWrapper:               jsonwrapper.NewDefaultJSONWrapper(),
 }
 
+// GetOverrideFxOptions provides a hook to configure a non zero number of fx options
+func GetOverrideFxOptions() []fx.Option {
+	return []fx.Option{}
+}
+
```

Comment:
I think adding a new field in zanzibar.Options is a more suitable way for passing such controls instead of exposing a new adhoc method. What do you think?

Reply:
Good point. I'd have liked that too there are two limiting reasons:
- fx is bound in the app prior to creating the gateway. 
- These options are more like Gateway options while we need a way to bind custom modules that may or may not be necessarily related to the gateway config.

Added some logs to share how FX is bound before gateway is created.
![image](https://user-images.githubusercontent.com/12872673/123317206-5fb6a180-d54b-11eb-8f3c-ccccc3ab0726.png)

Also, `go-autoprofiler` is also a candidate for a future fx module. It'd fit `GetOverrideFxOptions` instead of `zanzibar.Options`

Reply:
that is a fair point.